docs/libcurl/opts/CURLOPT_IPRESOLVE.3

```diff
@@ -29,14 +29,22 @@ CURLOPT_IPRESOLVE \- specify which IP protocol version to use
 CURLcode curl_easy_setopt(CURL *handle, CURLOPT_IPRESOLVE, long resolve);
 .SH DESCRIPTION
 Allows an application to select what kind of IP addresses to use when
-resolving host names. This is only interesting when using host names that
-resolve addresses using more than one version of IP. The allowed values are:
+resolving host names or establishing a connection. This is only interesting
+when using host names that resolve addresses using more than one version of
+IP. The allowed values are:
 .IP CURL_IPRESOLVE_WHATEVER
-Default, resolves addresses to all IP versions that your system allows.
+Default, resolves and uses addresses to all IP versions that your system
+allows.
 .IP CURL_IPRESOLVE_V4
-Resolve to IPv4 addresses.
+Resolve to IPv4 addresses only, but may use whatever is available in DNS cache
+or connection pool.
 .IP CURL_IPRESOLVE_V6
-Resolve to IPv6 addresses.
+Resolve to IPv6 addresses only, but may use whatever is available in DNS cache
+or connection pool.
```

Comment:
Yes, I agree, that is more or less what I said in [this comment](#issuecomment-814152151).

I see very little value in preserving the old behaviors of `CURL_IPRESOLVE_V6` (or `V4`), even for retrocompatibility, because this quirk of "maybe it is not really IPv6 that is used" is unreliable. But this is the documented behavior, as you pointed to me in the IRC channel once. From the docs: "Allows an application to select what kind of IP addresses to use when **resolving host names**."

I would prefer to only have the behavior of `_STRICT` variants. I initially did it in a new, separated setting `CURLOPT_IP_VERSION`, so that `CURLOPT_IPRESOLVE` could be eventually deprecated, and made as an alias the new option.

But if you prefer I simply replace the behavior of `CURLOPT_IPRESOLVE` with the new one, and update the documentation, it is fine by me, too. I only ask that you make this decision before I change this PR again.